src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,126 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private StorageDao storageDao;
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setUp() {
+        storageDao = mock(StorageDao.class);
+        registrationService = new RegistrationServiceImpl(storageDao);
+    }
```

Comment:
Consider using @BeforeAll instead of @BeforeEach if the initialization of StorageDao and RegistrationServiceImpl does not need to be done before each test. @BeforeAll is more efficient as it runs once before all tests, whereas @BeforeEach runs before each test method.